lib/bolt/plugin/module.rb

```diff
@@ -36,17 +36,17 @@ def initialize(mod:, context:, config:, **_opts)
       def setup
         @data = load_data
         @hook_map = find_hooks(@data['hooks'] || {})
-        # If there is a config section in bolt_plugin.json, validate against that and send
-        # validated values nested under `_config` key. Otherwise validate againsts the intersection
-        # of all task schemas.
-        # TODO: remove @send_config when deprecated
-        schema = if @data['config']
-                   @send_config = true
-                   @data['config']
-                 else
-                   extract_task_parameter_schema
-                 end
-        @config_schema = process_schema(schema)
+
+        if @data['config']
+          msg = <<~MSG.chomp
+            Found deprecated key 'config' in bolt_plugin.json. Config for a plugin is inferred
+            from task parameters, with config values passed as parameters.
+          MSG
+          raise InvalidPluginData.new(msg, name)
```

Comment:
It turns out every single use of this error class has the arguments in the reverse order, leading to messages like this:

```
Invalid Plugin Data for Found deprecated key 'config' in bolt_plugin.json. Config for a plugin is inferred
from task parameters, with config values passed as parameters.: test
```

Probably easier to swap the arguments in the initialize method than to update every caller. That's consistent with our other error classes which take `msg` as the first argument.